config/config/src/main/java/io/helidon/config/spi/ConfigParser.java

```diff
@@ -77,9 +78,25 @@
      * @param content a content to be parsed
      * @return parsed hierarchical configuration representation
      * @throws ConfigParserException in case of problem to parse configuration from the source
+     * @deprecated use {@link #parse(io.helidon.config.spi.ConfigParser.Content, java.util.function.Function)} instead
      */
+    @Deprecated
     ObjectNode parse(Content content) throws ConfigParserException;
 
+    /**
+     * Parses a specified {@link ConfigContent} into a {@link ObjectNode hierarchical configuration representation}.
+     * <p>
+     * Never returns {@code null}.
+     *
+     * @param content a content to be parsed
+     * @param relativeResolver function to resolve relative resources to this resource (if supported by the source)
```

Comment:
Can we add more javadoc to this?  Seems a bit opaque from the public reading of it.  Maybe an example.